### ISSUE TITLE
Provide an example KMP project showing the usage

### ISSUE THREAD

**Issue by makaronis** [Crossoid/Kotlin-Native-BigDecimal#2]
As the title says, it's not clear for me can i use it that way?

**Comment by kendy** [comment]
Do you mean if you can use this library so that in commonMain, you have code like 'import java.math.BigDecimal' and 'val x = BigDecimal(...)' etc., and can still build it for iOS?

Yes, being able to do that is the exact point of this library - when you build for Android, you will use the Java's java.math.BigDecimal, but on iOS, this library will implement exactly the same API - so you don't have to adapt your code in commonMain, or split it to androidMain and iosMain (and do different things for one or the other) or anything like that.

Let me close this issue for now, but please feel free to re-open if you asked something different. In that case, please explain me your question a bit more, ideally with code example.

**Comment by makaronis** [comment]
I have problems integrating this to my project. I've builded klibs for ios implementation, but how can i call java BigDecimal code from commonMain? Usage of java is forbidden in common package as i know

**Comment by kendy** [comment]
Normally it is forbidden, because there are no java.* packages to import when you compile for iOS.

But this project implements the java.math.BigDecimal even for iOS (see https://github.com/kendy/Kotlin-Native-BigDecimal/tree/master/src/iosMain/kotlin/java.math), so you can import java.math.BigDecimal in your commonMain just fine - on Android, it will import the native implementation, on iOS, it will use this project (when you use the klibs in the build.gradle.kts, as described in the README.md).

If you get any error messages though, please let me know!

**Comment by makaronis** [comment]
I don't understand how to use your project. I've builded klib files for ios implementation, but i don't get how to call that BigDecimal from common package. I think i need a to add a dependency like that to commonMain to use it, but i don't see anything related to that in readme. Could you please clarify that for me 
`val commonMain by getting {
            dependencies {
 implementation("kendy.BigDecimal")
            }
        }`

**Comment by kendy** [comment]
Nope, you don't need to add anything to commonMain - you never build a "commonMain", only the Android version of that (and there, the Android's BigDecimal is used), or the iOS version of that (and for that, you need to add the following snippet to shared/build.gradle.kts as described in the README.md):

```
val iosArm64Main by getting {
    dependencies {
        implementation(files("libs/iosArm64/BigDecimal.klib"))
        implementation(files("libs/iosArm64/BigDecimal-cinterop-boringssl.klib"))
    }
}
```

To avoid any misunderstanding, let me paste here a larger part of build.gradle.kts from my project:

```
    sourceSets {
        val commonMain by getting
        val commonTest by getting {
            dependencies {
                implementation(kotlin("test"))
            }
        }
        val androidMain by getting {
            dependencies {
                // all these are irrelevant for you - but it is a copy, so...
                implementation("com.google.android.material:material:1.2.1")
                implementation("androidx.core:core-ktx:+")
                implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version")
            }
        }
        val iosMain by getting {
        }
        val iosArm64Main by getting {
            dependsOn(iosMain)
            dependencies {
                // assuming you have copied the libraries to libs/iosArm64/
                implementation(files("libs/iosArm64/BigDecimal.klib"))
                implementation(files("libs/iosArm64/BigDecimal-cinterop-boringssl.klib"))
            }
        }
        val iosSimulatorArm64Main by getting {
            dependsOn(iosMain)
            dependencies {
                // assuming you have copied the libraries for simulator to libs/iosSimulatorArm64/
                implementation(files("libs/iosSimulatorArm64/BigDecimal.klib"))
                implementation(files("libs/iosSimulatorArm64/BigDecimal-cinterop-boringssl.klib"))
            }
        }
    }
```

In some Kotlin file in commonMain, you can try something like:

```
// on Android, this is built in, on iOS, this uses the Kotlin-Native-BigDecimal
import java.math.BigDecimal

fun something() {
    val bigOne = BigDecimal("1.0")
}
```

and see that it compiles both on Android and on iOS. In case you run into any build errors, please share the actual build errors of course.

In other words, when you've built the klibs, you have done all the hard work, now just profit :-)

Does this help please?

**Comment by makaronis** [comment]
I can't import java BigDecimal in my commonMain kotlin class( Maybe you can provide some sample project where it works ?

**Comment by kendy** [comment]
This is a great idea - I'll create a small example project so that it is easy for you & others to use - thanks!

Regarding "I can't import" - can you share the error message you are getting? Maybe we'll sort it out for you faster, before I get to creating the example.

**Comment by makaronis** [comment]
Thank you. About error, it just unresolved reference for "java"

**Comment by makaronis** [comment]
Any update on that ?

**Comment by elviss116** [comment]
please add example, is impossible import java.math in common class.

`iosMain.dependencies {
            implementation(files("libs/BigDecimal.klib"))
            implementation(files("libs/BigDecimal-cinterop-boringssl.klib"))
        }`

**Comment by kendy** [comment]
The example project is now available here: https://github.com/Crossoid/Kotlin-Native-BigDecimal-Example

Also I started providing binary packages, you don't have to compile the klibs on your own any more :-)

Hope this helps, please let me know should you have any questions!

**Comment by davida5** [comment]
I've been having this same issue `java.` not resolving in common, and installed the sample app, which prompted me to update gradle and after update java isn't resolving:
<img width="913" alt="image" src="https://github.com/user-attachments/assets/f88865bc-f07b-40c8-8872-cf04d0b18022" />

**Comment by mak1nt0sh** [comment]
Same, followed the example app and not resolving BigDecimal.
